src/app/modules/+badges/achievements-list/achievements-list.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+
+import { RequestDataService } from './../../../shared/services/request-data.service';
+
+import { Achievement } from '../../../shared/models/achievement';
+
+@Component({
+  selector: 'app-achievements-list',
+  templateUrl: './achievements-list.component.html',
+  styleUrls: ['./achievements-list.component.scss']
+})
+export class AchievementsListComponent implements OnInit {
+  achievements$: Observable<Achievement[]>;
+  displayedColumns: string[];
+
+  constructor(private requestDataService: RequestDataService) { }
+
+  ngOnInit(): void {
+    this.setAllAchievements();
+    this.displayedColumns = ['title', 'xp', 'date'];
+  }
+
+  setAllAchievements(): void {
```

Comment:
I would rename this function to `fetchAllAchievements`. But it is up to you :)

Reply:
I assume in service I get data and set it to variable in component. But I think method in service maybe better call with 'fetch':)
Interesting to hear your opinion